### PR TITLE
Vulkan: Use VK_NV_glsl extension where available, and skip glslang

### PR DESCRIPTION
**Pull request by stenzek**
Provides a speedup in shader compilation time compared to going through glslang.

We had this previously enabled but had to revert it because dynamic sampler indexing was broken somewhere between 384.xx and 385.xx. It is now fixed in 387.xx.

Should let us close these issues. I'm still not sure why the glslang path breaks on some systems but not others, but using the extension doesn't seem to have this problem.

https://bugs.dolphin-emu.org/issues/10527
https://bugs.dolphin-emu.org/issues/10301
https://bugs.dolphin-emu.org/issues/10224
https://bugs.dolphin-emu.org/issues/10535

### CODE REVIEW

File: Source/Core/VideoBackends/Vulkan/VulkanContext.cpp [hunk 1]
```diff
@@ -426,6 +432,7 @@ bool VulkanContext::SelectDeviceExtensions(ExtensionList* extension_list, bool e
   if (enable_surface && !CheckForExtension(VK_KHR_SWAPCHAIN_EXTENSION_NAME, true))
     return false;
 
+  CheckForExtension(VK_NV_GLSL_SHADER_EXTENSION_NAME, false, &m_supports_nv_glsl_extension);
```
Comment (riking) [inline Source/Core/VideoBackends/Vulkan/VulkanContext.cpp]: Why the change to the function when this would work just as well?

```
m_supports_nv_glsl_extension = CheckForExtension(VK_NV_GLSL_SHADER_EXTENSION_NAME, false);
```

**Comment by psennermann** [comment]
Any particular reason to keep this on hold? From what I undestand it's all about reverting to the behaviour previous to the Nvidia bugged Driver...also it seems that without this PR realXFB and Vertex Rounding won't ever work right in Vulkan

**Comment by stenzek** [comment]
It's just waiting on review/approval.

**Comment by psennermann** [comment]
Everyone's on holiday? :-)

**Comment by tabnk** [comment]
Very useful for Vulkan. Waiting for merge :)

**Comment by LuismaSP89** [comment]
@stenzek When is this going to be merged? (Asking only for curiosity, i don´t want to be impertinent)

**Comment by aunali1** [comment]
Just tested on Pixel C (NVIDIA Tegra X1), seems to reduce ubershader compile times by a few seconds. Everything seems to work fine AFAICT.

**Comment by LuismaSP89** [comment]
@Helios747 Thanks for your censhorship to me and other two users only for asking something with education. Now the world is a better place, and we owe everything to you ; )

**Comment by psennermann** [comment]
Yes the compilation is slightly faster, but what's much more important is that it finally fixes all those issues...
